samples/sample_encode/src/pipeline_encode.cpp

```diff
@@ -2408,12 +2440,31 @@ mfxStatus CEncodingPipeline::LoadNextFrame(mfxFrameSurface1* pSurf)
         }
     }
     // frameorder required for reflist, dbp, and decrefpicmarking operations
-    if (pSurf) pSurf->Data.FrameOrder = m_nFramesRead;
+    if (pSurf)
+        pSurf->Data.FrameOrder = m_mfxEncParams.mfx.EncodedOrder ? m_QPFileReader.GetCurrentDisplayOrder() : m_nFramesRead;
     m_nFramesRead++;
 
     return sts;
 }
 
+mfxStatus CEncodingPipeline::LoadNextControl(mfxEncodeCtrlWrap*& pCtrl, mfxU32 encSurfIdx)
+{
+    try
+    {
+        if (!pCtrl) return MFX_ERR_NULL_PTR;
```

Comment:
now it useless check, since you are setting control in any way on the next line

Reply:
removed it, possible errors will be catched by catch